airflow/utils/log/log_reader.py

```diff
@@ -77,12 +78,16 @@ def read_log_stream(self, ti: TaskInstance, try_number: int | None, metadata: di
             metadata.pop("max_offset", None)
             metadata.pop("offset", None)
             metadata.pop("log_pos", None)
-            while "end_of_log" not in metadata or (
-                not metadata["end_of_log"] and ti.state not in [State.RUNNING, State.DEFERRED]
-            ):
+            while True:
                 logs, metadata = self.read_log_chunks(ti, current_try_number, metadata)
                 for host, log in logs[0]:
                     yield "\n".join([host or "", log]) + "\n"
+                if "end_of_log" not in metadata or (
+                    not metadata["end_of_log"] and ti.state not in [State.RUNNING, State.DEFERRED]
+                ):
+                    time.sleep(0.5)
```

Comment:
why 0.5 ?

Reply:
looks like a magic number ...

Reply:
Completely as arbitrary

Reply:
You prefer one second, perhaps?

Reply:
yeah looking again, i think better would be to do something more akin to exponential backoff... i.e. if in two consecutive loops there are no logs, then sleep 0.5. otherwise let it run.

previously i had this only sleep while waiting before getting _any_ logs.  i.e. when offset stays at 0.

Reply:
ok, implemented more selective throttling, added comments, made it overrideable through airflow local settings bby using constant